### PR TITLE
MJX Training Implementation

### PR DESCRIPTION
**Pull request by michael-lutz**
The following is copied from https://github.com/kscalelabs/sim/pull/14

This PR introduces a new way to massively scale up locomotion training. Building upon Brax, it ultimately uses the MJX physics engine for simulation.

### Structure
Specifically, this PR includes the following directories:

- Envs
- Experiments
- Utils
- (example) Weights
- train.py
- play.py

`Envs` includes two types of Brax environments: DefaultHumanoidEnv and StompyEnv. Each environment includes a main class which implements the Brax environment interface and utilizes MJX for all physics calculations. One important thing to note is that reward functions are **modular**, allowing for quick experimentation.

`Experiments` includes two .yaml files that include sample configurations for model training.

`Utils` include default values, rendering rollouts, etc.

`Weights` currently include default humanoid weights (for locomotion) that should work out of the box

train.py and play.py both integrate with wandb. train.py utilizes the Brax implementation of PPO for now, but can be easily customized if needed.

### Performance Samples
**Training Curves**
Screen Shot 2024-05-22 at 9 12 45 PM
Screen Shot 2024-05-22 at 9 14 44 PM

**Example humanoid robot walking in MJX**
https://github.com/kscalelabs/sim/assets/43460304/8e12b0e6-48ea-4af0-8283-1dc4880767b4

**Humanoid trained in MJX, eval in CPU-based MuJoCo**
https://github.com/kscalelabs/sim/assets/43460304/7f158aeb-6bc9-4056-bd1d-12882adbd13c

### CODE REVIEW

File: ksim/mjx_gym/envs/__init__.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from brax import envs
```
Comment (budzianowski) [inline ksim/mjx_gym/envs/__init__.py:3]: If you can omit putting stuff here, that would be preferable.

File: ksim/mjx_gym/envs/default_humanoid_env/default_humanoid.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
+import jax
```
Comment (budzianowski) [inline ksim/mjx_gym/envs/default_humanoid_env/default_humanoid.py:5]: Adding one line explanation would be useful.

File: ksim/mjx_gym/envs/default_humanoid_env/default_humanoid.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+import jax.numpy as jp
```
Comment (budzianowski) [inline ksim/mjx_gym/envs/default_humanoid_env/default_humanoid.py:6]: isort

File: ksim/mjx_gym/envs/default_humanoid_env/default_humanoid.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        """Observes humanoid body position, velocities, and angles.
+
+        Args:
+            data: The current state of the environment.
+            action: The current action.
+        Returns:
+            Observations of the environment.
+        """
+        position = data.qpos
+        if self._exclude_current_positions_from_observation:
```
Comment (budzianowski) [inline ksim/mjx_gym/envs/default_humanoid_env/default_humanoid.py:163]: Why this is needed?

**Review by budzianowski — APPROVED** [review]
A couple of nits, lgtm otherwise!

**Review by codekansas — APPROVED** [review]
lgtm
